src/components/SignUpForm/index.tsx

```diff
@@ -0,0 +1,318 @@
+  const [mbti, setMbti] = useState<string>('1');
+  const [gender, setGender] = useState<string>('1');
+  // Valid
+  const [isValidLoginId, setIsValidLoginId] = useState<boolean>(false);
+  const [isValidNickname, setIsValidNickname] = useState<boolean>(false);
+  const [isValidPassword, setIsValidPassword] = useState<boolean>(false);
+  const [isValidPasswordConfirm, setIsValidPasswordConfirm] = useState<boolean>(false);
+  const [isValidEmail, setIsValidEmail] = useState<boolean>(false);
+  const [isValidMbti, setIsValidMbti] = useState<boolean>(false);
+  // Error Message
+  const [loginIdErrorMessage, setLoginIdErrorMessage] = useState<string>(ID_REQUIRE_CHECK); // 추후 아이디 중복 체크시 에러메시지 useState
+  const [nicknameErrorMessage, setNickNameErrorMessage] = useState<string>(NICKNAME_REQUIRE_CHECK);
+  const [passwordErrorMessage, setPasswordErrorMessage] = useState<string>(PW_REQUIRE_CHECK);
+  const [passwordConfirmErrorMessage, setPasswordConfirmErrorMessage] =
+    useState<string>(PWCONFIRM_REQUIRE_CHECK);
+  const [emailErrorMessage, setEmailErrorMessage] = useState<string>(EMAIL_REQUIRE_CHECK);
+  const [mbtiErrorMessage, setMbtiErrorMessage] = useState<string>(MBTI_REQUIRE_CHECK);
+
+  // MBTI select option
+  const mbtis: Mbti[] = [
+    { id: '1', name: '선택하세요' },
+    { id: '2', name: 'INFJ' },
+  ];
+
+  // Gender select option
+  const genders: Gender[] = [
+    { id: '1', name: '남' },
+    { id: '2', name: '여' },
+  ];
+
+  // LoginId Focus : ref styled component ISSUE
+  // const loginIdFocus = useCallback((inputElement) => {
+  //   if (inputElement) {
+  //     inputElement.focus();
+  //   }
+  // }, []);
+
+  // LoginId Change Event
+  const onChangeLoginId = (e: SyntheticEvent<HTMLInputElement>) => {
+    setLoginId(e.currentTarget.value);
+    setIsValidLoginId(false);
+    // console.log(`onChange :${isValidLoginId}`);
+  };
+
+  // Nickname Change Event
+  const onChangeNickname = (e: SyntheticEvent<HTMLInputElement>) => {
+    setNickname(e.currentTarget.value);
+    setIsValidNickname(false);
+  };
+
+  // Password Change Event
+  const onChangePassword = (e: SyntheticEvent<HTMLInputElement>) => {
+    setPassword(e.currentTarget.value);
+    setIsValidPassword(false);
+  };
+
+  // PasswordConfirm Change Event
+  const onChangePasswordConfirm = (e: SyntheticEvent<HTMLInputElement>) => {
+    setPasswordConfirm(e.currentTarget.value);
+    setIsValidPasswordConfirm(false);
+  };
+
+  // Email Change Event
+  const onChangeEmail = (e: SyntheticEvent<HTMLInputElement>) => {
+    setEmail(e.currentTarget.value);
+    setIsValidEmail(false);
+  };
+
+  // Mbti Change Event
+  const onChangeMbti = (e: SyntheticEvent<HTMLSelectElement>) => {
+    setMbti(e.currentTarget.value);
+    if (e.currentTarget.value !== '1') {
+      setIsValidMbti(false);
+    }
+  };
+
+  // Gender Change Event
+  const onChangeGender = (e: SyntheticEvent<HTMLSelectElement>) => {
+    setGender(e.currentTarget.value);
+  };
```

Comment:
```suggestion
const handleChange = (e: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>) => {
  const { name, value } = e.target;
  setState(prevState => ({ ...prevState, [name]: value }));
  setValidationState(prevState => ({ ...prevState, [name]: { isValid: false, errorMessage: validationMessages[name] } }));
};
```

위에서 의견드린 state대로 변경하신다면 긴 코드를 한개의 함수로 컨트롤 할 수 있게됩니다.